src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        try {
+            String[] salaryWorkers = new String[names.length];
+            String pattern = "dd.MM.yyyy";
+            SimpleDateFormat sdf = new SimpleDateFormat(pattern);
+            Date startWork = sdf.parse(dateFrom);
+            Date finishWork = sdf.parse(dateTo);
+            String[] returnStrings = new String[names.length];
+            int[] sums = new int[names.length];
+            for (int a = 0; a < names.length; a++) {
+                String someName = names[a];
```

Comment:
someName - not really informative looks like you just forgot to put the name of variable, nameAtIndex etc. will be much more obvious